mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -33,36 +33,34 @@ import org.apache.spark.ml.classification.{OneVsRest, OneVsRestModel}
 import org.apache.spark.ml.feature.RFormulaModel
 import org.apache.spark.ml.param.{ParamPair, Params}
 import org.apache.spark.ml.tuning.ValidatorParams
-import org.apache.spark.sql.{SparkSession, SQLContext}
+import org.apache.spark.sql.SparkSession
 import org.apache.spark.util.Utils
 
 /**
  * Trait for [[MLWriter]] and [[MLReader]].
  */
 private[util] sealed trait BaseReadWrite {
-  private var optionSQLContext: Option[SQLContext] = None
+  private var optionSparkSession: Option[SparkSession] = None
 
   /**
-   * Sets the SQL context to use for saving/loading.
+   * Sets the Spark Session to use for saving/loading.
    */
   @Since("1.6.0")
-  def context(sqlContext: SQLContext): this.type = {
-    optionSQLContext = Option(sqlContext)
+  def context(sparkSession: SparkSession): this.type = {
```

Comment:
I'd prefer to maintain compatibility.  Can you create a new method which accepts a SparkSession?